README.md

```diff
@@ -25,12 +35,276 @@ Three new interfaces are introduced as part of the API:
-  - `Promise<PrintJob> printJob(...)` implements the `Print-Job` IPP attribute and allows developers to customize the print job via a subset of supported job template attributes specified in section [5.2 of RFC8011](https://www.rfc-editor.org/rfc/rfc8011#section-5.2).
-  - `readonly attribute PrinterState state` and `attribute EventHandler onstatechange` allows developers to track the printer state and wait until it becomes ready to accept print jobs.
+  - `Promise<void> updateAttributes()` implements the `Get-Printer-Attributes` IPP operation that updates `readonly attribute PrinterAttributes attributes` and returns a promise that is settled once the operation completes. The subset of supported printer attributes specified in section [5.4 of RFC8011](https://www.rfc-editor.org/rfc/rfc8011#section-5.4).
+  - `Promise<PrintJob> printJob(...)` implements the `Print-Job` IPP operation and allows developers to customize the print job via a subset of supported job template attributes specified in section [5.2 of RFC8011](https://www.rfc-editor.org/rfc/rfc8011#section-5.2).
 - The `PrintJob` interface provides a way to interact with print jobs: monitor their state changes and cancel on demand.
   - `void cancel()` can be invoked to attempt to cancel a print job.
-  - `readonly attribute PrintJobState state` and `attribute EventHandler onstatechange` allows developers to track the job state and wait for its completion.
+  - `readonly attribute PrintJobAttributes attributes` (and dictionary fields like `job-state`, `job-state-message` and `job-state-reasons`) and `attribute EventHandler onjobstatechange` allow developers to track the job state and wait for its completion.
+
+## IPP Mapping
+
+Attribute names/values in the proposed API are modelled with respect to the existing IPP names as specified in [Internet Printing Protocol (IPP) Registrations](https://www.iana.org/assignments/ipp-registrations/ipp-registrations.xhtml).
+
+The following table specifies WebIDL type mapping for selected types listed in section [5.1 of RFC8011](https://www.rfc-editor.org/rfc/rfc8011#section-5.1).
+
+| IPP Type | WebIDL Type |
+| :-------------: | :-------------: |
+| `text` | `DOMString` |
+| `name` | `DOMString` |
+| `keyword` | `enum` |
+| `enum` | `enum` |
+| `keyword` \| `name` | `DOMString` |
+| `uri` | `DOMString` |
+| `boolean` | `boolean` |
+| `integer` | `unsigned long` |
+| `1setOfX` | `sequence<X>` |
+| `rangeOfInteger` | `dictionary { unsigned long from, unsigned long to };` |
+| `octetString` | `DOMString` |
+| `resolution` | `dictionary {`<br>`unsigned long cross-feed-direction-resolution,`<br>`unsigned long feed-direction-resolution,`<br> `PrintingResolutionUnits units`<br>`};` |
+| `collection` | `dictionary { ... }` |
+
+Attribute names retain their hyphen-case notation as `dictionary` members.
```

Comment:
Can you justify this over the alternative of translating attribute names to lower-camelcase to match WebIDL conventions? I believe this is going to make it difficult to access these properties because `mediaSize.x-dimension` will be parsed as `mediaSize.x - dimension` so developers will have to write `mediaSize['x-dimension']` instead.

Reply:
Have you validated that existing WebIDL machinery will work with these field names?

Reply:
Just to confirm, it looks like the WebIDL grammar definition for [identifier](https://webidl.spec.whatwg.org/#prod-identifier) allows attributes to contain hyphens: `/[_-]?[A-Za-z][0-9A-Z_a-z-]*/`

Reply:
I believe we should be safe here then.

On a side note, I'm wondering what the point of the preceding `-` (from `[_-]?`) is -- does WebIDL also allow things like `-kebab-case-name`?

Reply:
I believe so. Such names have been used for vendor prefixed attributes.